.travis.yml

```diff
@@ -56,20 +58,25 @@ jobs:
         # Ensure that we build without warnings.
         - CARGO_FLAGS="--features 'strict'"
         - TRAVIS_WITH_NO_WINDOW_SYSTEM=1  # So we can tell them apart
-        - TRAVIS_ALLOW_FAILURE=1
+        #- TRAVIS_ALLOW_FAILURE=1
+      before_script:
+        - travis_wait rustup install $(cat rust_src/rust-toolchain)
+        - rustup default $(cat rust_src/rust-toolchain)
       script:
         - ./autogen.sh && ./configure --without-makeinfo --with-x=no --with-ns=no --without-gconf --without-gsettings
         - make -j 3 && echo '==> Running nowindow tests' && make check
     -
       <<: *NO_WINDOW_SYSTEM_TEST
       os: osx
       before_script:
+        - travis_wait rustup install $(cat rust_src/rust-toolchain)
+        - rustup default $(cat rust_src/rust-toolchain)
         # Travis docs say to avoid brew update, but it works around the incorrect Ruby version
         - brew update
         - brew install gnutls
         - brew install texinfo
-  allow_failures:
-    - env: TRAVIS_ALLOW_FAILURE=1
+  #allow_failures:
+  #  - env: TRAVIS_ALLOW_FAILURE=1
```

Comment:
Can we remove this?

Reply:
I left it in to show people how to enable it is we need it in the future. With the stages system it took me a bit to figure out how to allow for expected failure. The current Travis docs are not written for stages.

I can remove it or leave it. Do you want more comments? Move it out to a comment block? Put it in a document like say contributing?

Reply:
I'd prefer that we remove it from here -- there's a risk that commented code bitrots. Putting in the contributing guidelines sounds like a great idea.

Reply:
Doc updated.